### PR TITLE
Intake

### LINKED ISSUE #3: Intake
`VictorSPX rollerMotor`
`DoubleSolenoid lifterPiston`

### PR DESCRIPTION
**Pull request by TapChap**
fixes #3

### CODE REVIEW

**Comment by Starlight220** [comment]
@elon698 please review this

File: src/main/java/io/excaliburfrc/robot/subsystems/Intake.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      intakeMotor.set(speed.speed);
+    } else {
+      intakeMotor.set(Mode.OFF.speed);
+    }
+  }
+
+  public void stop() {
+    intakeMotor.set(Mode.OFF.speed);
+  }
+}
```
Comment (elon698) [inline src/main/java/io/excaliburfrc/robot/subsystems/Intake.java]: havnt found any bugs here

**Review by elon698 — APPROVED** [review]

File: src/main/java/io/excaliburfrc/robot/subsystems/Intake.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    public final double speed;
+  }
+
+  public void activate(Mode speed) {
+    if (piston.get() == DoubleSolenoid.Value.kReverse) {
+      intakeMotor.set(speed.speed);
+    } else {
+      intakeMotor.set(Mode.OFF.speed);
+    }
+  }
```
Comment (Starlight220) [inline src/main/java/io/excaliburfrc/robot/subsystems/Intake.java]: Shouldn't this be the other way around? (do nothing if closed)
Also, I suggest adding a comment that explains that.

**Review by Starlight220 — CHANGES_REQUESTED** [review]
Other than the logic (and comment) issue, there are missing functions (and button definitions) for dealing with the piston. Assume that the piston is "up"/"open" on `kForward`.

File: src/main/java/io/excaliburfrc/robot/Constants.java [hunk 3]
```diff
@@ -10,6 +10,12 @@
  */
 public final class Constants {
   // add a inner class `public static final class` for each subsystem.
+  public static final class IntakeConstants {
+    public static final int INTAKE_MOTOR_ID = 0;
+    public static final int FORWARD_CHANNEL = 0;
+    public static final int REVERSE_CHANNEL = 1;
```
Comment (Starlight220) [inline src/main/java/io/excaliburfrc/robot/Constants.java]: Real ports are: 
```
motor id = 60
fwd = 1
rev =  0
``` 
Don't change the constant names though - only the port numbers.

**Review by Starlight220 — DISMISSED** [review]

**Review by TapChap — COMMENTED** [review]
and fixed port numbers

File: src/main/java/io/excaliburfrc/robot/subsystems/Intake.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    public final double speed;
+  }
+
+  public void raisePisto() {
+    piston.set(DoubleSolenoid.Value.kForward);
+  }
+
+  public void lowerPisto() {
+    piston.set(DoubleSolenoid.Value.kReverse);
+  }
```
Comment (Starlight220) [inline src/main/java/io/excaliburfrc/robot/subsystems/Intake.java]: why the duplicates?

File: src/main/java/io/excaliburfrc/robot/RobotContainer.java [hunk 5]
```diff
@@ -28,11 +30,34 @@ public RobotContainer() {
 
   /**
    * Use this method to define your button->command mappings. Buttons can be created by
-   * instantiating a {@link GenericHID} or one of its subclasses ({@link
-   * edu.wpi.first.wpilibj.Joystick} or {@link XboxController}), and then calling passing it to a
-   * {@link JoystickButton}.
+   * instantiating a {@link } or one of its subclasses ({@link edu.wpi.first.wpilibj.Joystick} or
```
Comment (Starlight220) [inline src/main/java/io/excaliburfrc/robot/RobotContainer.java]: just delete the whole comment, don't leave it like this

File: src/main/java/io/excaliburfrc/robot/RobotContainer.java [hunk 6]
```diff
@@ -28,11 +30,34 @@ public RobotContainer() {
...
+        .whenPressed(
+            () -> {
+              intake.raisePisto();
+            },
+            intake)
+        .whenReleased(
+            () -> {
+              intake.lowerPisto();
+            },
+            intake);
```
Comment (Starlight220) [inline src/main/java/io/excaliburfrc/robot/RobotContainer.java]: split this to two buttons: one for opening and one for closing.

**Review by Starlight220 — CHANGES_REQUESTED** [review]

File: src/main/java/io/excaliburfrc/robot/subsystems/Intake.java (hunk 4 above)
Comment (korozya-bit) [reply src/main/java/io/excaliburfrc/robot/subsystems/Intake.java:37]: what do you mean why? one to raise and one to lower, also that's what you told me to do

**Review by Starlight220 — APPROVED** [review]

**Comment by Starlight220** [comment]
Rebase, and I'll squash this into `main`.

**Pull request merged**